### PR TITLE
Experiment file with the dendritic network

### PR DESCRIPTION
**Pull request by lucasosouza**
No changes to the framework (except comments), all changes encapsulated within the dendrites.py experiment file.

### CODE REVIEW

**Comment by lucasosouza** [comment]
I ended up commiting some of the TODO comments to the docstring we did on the pair programming session, since I need to push the branch to github. If you prefer I can remove before merging.

File: projects/meta_cl/experiments/dendrites.py [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    # Number of tasks (i.e. distinct classes) for the inner loops and outer loop
+    tasks_per_epoch=3,
+
+    # Batch size for each inner and outer step, respectively.
+    batch_size=1,       # use 1 images per step w/in the inner loop
+    slow_batch_size=5,  # use 5 images per class w/in the outer loop.
+
+    # Number of steps per task within the inner loop.
+    num_fast_steps=5,  # => we take 5 * 'tasks_per_epoch' sequential grad steps
+    # num_slow_steps=1,  # not actually defined, but technically this is always one.
```
Comment (karangrewal) [inline projects/meta_cl/experiments/dendrites.py:201]: If we don't need `num_slow_steps`, then let's remove it.

File: projects/meta_cl/experiments/dendrites.py [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+from nupic.research.frameworks.vernon import MetaContinualLearningExperiment, mixins
+from nupic.research.frameworks.vernon.trainables import SupervisedTrainable
+
+
+class OMLExperiment(mixins.OnlineMetaLearning,
+                    MetaContinualLearningExperiment):
+    pass
+
+
+class DendriticNetwork(nn.Module):
```
Comment (karangrewal) [inline projects/meta_cl/experiments/dendrites.py:39]: Can you add a few sentences about this class/network? That may be helpful in the future.

**Review by karangrewal — APPROVED** [review]
Looks good. I left a few comments.

File: projects/meta_cl/experiments/dendrites.py (hunk 1 above)
Comment (lucasosouza) [reply projects/meta_cl/experiments/dendrites.py:201]: I would generally agree if it was not for an experiments file. We are usually ok with commented code in experiments file. It serves as documentation, and it is common that you are switching things on and off during experiments.

File: projects/meta_cl/experiments/dendrites.py (hunk 2 above)
Comment (lucasosouza) [reply projects/meta_cl/experiments/dendrites.py:39]: Added

File: projects/meta_cl/experiments/dendrites.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+    exp.setup_experiment(config)
+    print(f"Training started....")
+    while not exp.should_stop():
+        result = exp.run_epoch()
+        print(f"Accuracy: {result['mean_accuracy']:.4f}")
+    print(f"....Training finished")
+
+
+# Simple Omniglot Experiment
+metacl_base = dict(
```
Comment (mvacaporale) [inline projects/meta_cl/experiments/dendrites.py:160]: Can you remove the unused configs?

**Review by mvacaporale — APPROVED** [review]
Looks good. I can address the TODO's once you merge. I'll also clean up the code a bit (say, moving the `DendriticNetwork` into a `projects/meta_cl/networks` folder). Just a heads up.

File: projects/meta_cl/experiments/dendrites.py (hunk 3 above)
Comment (lucasosouza) [reply projects/meta_cl/experiments/dendrites.py:160]: There are no unused configs in this file. Each one is copying from the previous one. Notice the only one exposed though is the metacl_dendrites (ps: I didn't want it to have dependencies on other files)

Comment (mvacaporale) [reply projects/meta_cl/experiments/dendrites.py:160]: Oh I see

**Pull request merged**